components/form-builder/app/navigation/EditNavigation.tsx

```diff
@@ -20,32 +20,51 @@ export const EditNavigation = () => {
   };
 
   return (
-    <div className="relative">
+    <div className="relative flex">
       <nav className="mb-8 flex divide-x-2 divide-gray-600" aria-label={t("navLabelEditor")}>
         <SubNavLink href="/form-builder/edit">{t("questions")}</SubNavLink>
         <SubNavLink href="/form-builder/edit/translate">{t("translate")}</SubNavLink>
       </nav>
       {activePathname.endsWith("/edit") && (
-        <div className="absolute right-0 mr-24 top-0">
-          <label htmlFor="lang" className="font-bold text-sm">
+        <div className="absolute right-0 mr-24 top-0 flex items-baseline">
+          <span id="switch-heading" className="font-bold text-sm mr-1" aria-hidden="true">
             {t("editingIn")}{" "}
-          </label>
-          <label htmlFor="lang" className="text-sm">
-            {t("English")}
-          </label>{" "}
-          {translationLanguagePriority === "fr" && (
-            <button id="lang" onClick={() => switchLang()}>
-              <ToggleRight className="inline-block w-12 fill-fuchsia-300" />
-            </button>
-          )}
-          {translationLanguagePriority === "en" && (
-            <button id="lang" onClick={() => switchLang()}>
+          </span>
+          <div
+            role="button spinbutton"
```

Comment:
This is a hack :) So...... a button cannot use the aria-activedescendant property which is used to announce state changes. However aria allows multiple roles to be assigned to an element. A browser will read and probably only pay attention to the first role, current behavior. The second problem involving automated WCAG test tools, the second role will be picked up by automated tools (or just confuse it with two roles) and allow the aria-activedescendant on a spinbutton role. So it works but it-is-not-pretty :)